pkg/router/template/util/util.go

```diff
@@ -51,6 +51,12 @@ func GenerateRouteRegexp(hostname, path string, wildcard bool) string {
 		subpathRE = "(/.*)?"
 	}
 
+	// The path could contain space characters, which must be escaped in
```

Comment:
Could the host also contain space characters?

Reply:
No, the API would reject it:
```
% oc -n openshift-ingress-canary create route edge test-route-with-host-with-space --service=ingress-canary --hostname='www.foo bar.com'
The Route "test-route-with-host-with-space" is invalid:
* spec.host: Invalid value: "www.foo bar.com": host must conform to DNS 952 subdomain conventions
* spec.host: Invalid value: "www.foo bar.com": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')                                                                                                                                                
zsh: exit 1     oc -n openshift-ingress-canary create route edge  --service=ingress-canary
```